batch.go

```diff
@@ -19,8 +21,12 @@ type (
 
 	// EventBatchIterator provides an easy way to iterate over a slice of events to reliably create batches
 	EventBatchIterator struct {
-		Cursors            map[string]int
-		PartitionEventsMap map[string][]*Event
+		// Current chunk number
+		Cursor int
+		// Current position in the current chunk
+		PartitionCursor int
+		// Events grouped by the same partition key
+		PartitionEventChunks [][]*Event
```

Comment:
These are all breaking changes. Is it possible to use Cursors and PartitionEventsMap as they are now? Then add any additional fields to the struct as necessary?

Reply:
I switched back to the original data structure. The only major change I made is to return with an error message if event is bigger than the max batch size and I step the iterator over (not having the infinite loop). Also I put the sub-partition fix what I found a previous PR.